diff -ruN wpeframework-plugins.orig/WebKitBrowser/WebKitImplementation.cpp wpeframework-plugins/WebKitBrowser/WebKitImplementation.cpp
--- wpeframework-plugins.orig/WebKitBrowser/WebKitImplementation.cpp	2020-09-10 13:56:36.000000000 +0530
+++ wpeframework-plugins/WebKitBrowser/WebKitImplementation.cpp	2020-09-14 18:51:39.047344504 +0530
@@ -1321,7 +1321,7 @@
             // Oke, so we are good to go.. Release....
             Core::Thread::Run();
 
-            _configurationCompleted.WaitState(true, Core::infinite);
+            // _configurationCompleted.WaitState(true, Core::infinite);
 
             return (Core::ERROR_NONE);
         }
@@ -1667,7 +1667,7 @@
             g_signal_connect(_view, "permission-request", reinterpret_cast<GCallback>(decidePermissionCallback), nullptr);
             g_signal_connect(_view, "show-notification", reinterpret_cast<GCallback>(showNotificationCallback), this);
 
-            _configurationCompleted.SetState(true);
+            // _configurationCompleted.SetState(true);
 
             URL(static_cast<const string>(_URL));
 
