diff --git a/source/app/libpd.c b/source/app/libpd.c
index 08403a2..e4b151e 100644
--- a/source/app/libpd.c
+++ b/source/app/libpd.c
@@ -496,44 +496,43 @@ void libpd_client_mgr()
 
 static void get_parodus_url(char **parodus_url, char **client_url)
 {
+	const char* envDataFile = getenv("DATA_FILE");
 
-	FILE *fp = fopen(DEVICE_PROPS_FILE, "r");
+	const char* dataFile = ((NULL == envDataFile)? strdup(DEVICE_PROPS_FILE): strdup(envDataFile));
+	FILE *fp = fopen(dataFile, "r");
 
 	if (NULL != fp)
 	{
 		char str[255] = {'\0'};
 		while(fscanf(fp,"%s", str) != EOF)
 		{
-		    char *value = NULL;
+			char *value = NULL;
 
-		    if(NULL != (value = strstr(str, "PARODUS_URL=")))
-		    {
-			value = value + strlen("PARODUS_URL=");
-			*parodus_url = strdup(value);
-		    }
+			if(NULL != (value = strstr(str, "PARODUS_URL=")))
+			{
+				value = value + strlen("PARODUS_URL=");
+				*parodus_url = strdup(value);
+			}
 
-		    if(NULL != (value = strstr(str, "WEBPA_CLIENT_URL=")))
-		    {
-			value = value + strlen("WEBPA_CLIENT_URL=");
-			*client_url = strdup(value);
-		    }
+			if(NULL != (value = strstr(str, "WEBPA_CLIENT_URL=")))
+			{
+				value = value + strlen("WEBPA_CLIENT_URL=");
+				*client_url = strdup(value);
+			}
 
 		}
 		fclose(fp);
 	}
-	else
-	{
-		WalError("Failed to open device.properties file:%s\n", DEVICE_PROPS_FILE);
-		WalInfo("Adding default values for parodus_url and client_url\n");
-		*parodus_url = strdup(PARODUS_URL_DEFAULT);
-		*client_url = strdup(CLIENT_URL_DEFAULT);
-	}
+	free(dataFile);
 
 	if (NULL == *parodus_url)
 	{
 		WalError("parodus_url is not present in device.properties, adding default parodus_url\n");
-		
-		*parodus_url = strdup(PARODUS_URL_DEFAULT);
+		*parodus_url = getenv("PARODUS_URL");
+		if (NULL == parodus_url)
+		{
+			*parodus_url = strdup(PARODUS_URL_DEFAULT);
+		}
 	}
 	else
 	{
@@ -543,7 +542,11 @@ static void get_parodus_url(char **parodus_url, char **client_url)
 	if (NULL == *client_url)
 	{
 		WalError("client_url is not present in device.properties, adding default client_url\n");
-		*client_url = strdup(CLIENT_URL_DEFAULT);
+		*client_url = getenv("PARODUS2CCSP_CLIENT_URL");
+		if (NULL == client_url)
+		{
+			*client_url = strdup(CLIENT_URL_DEFAULT);
+		}
 	}
 	else
 	{
