diff --git a/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp b/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
index b49d47a9277c..c36f6b940ff9 100644
--- a/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
@@ -283,12 +283,12 @@ public:
 #if USE(GSTREAMER_GL)
     virtual void waitForCPUSync()
     {
-        GstGLSyncMeta* meta = gst_buffer_get_gl_sync_meta(m_buffer.get());
-        if (meta) {
-            GstMemory* mem = gst_buffer_peek_memory(m_buffer.get(), 0);
-            GstGLContext* context = ((GstGLBaseMemory*)mem)->context;
-            gst_gl_sync_meta_wait_cpu(meta, context);
-        }
+        // GstGLSyncMeta* meta = gst_buffer_get_gl_sync_meta(m_buffer.get());
+        // if (meta) {
+        //     GstMemory* mem = gst_buffer_peek_memory(m_buffer.get(), 0);
+        //     GstGLContext* context = ((GstGLBaseMemory*)mem)->context;
+        //     gst_gl_sync_meta_wait_cpu(meta, context);
+        // }
     }
 #endif // USE(GSTREAMER_GL)
 
