From cf2d36cfee234b11390e69b52d96409371f3bd51 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Krause?= <joerg.krause@embedded.rocks>
Date: Tue, 21 Mar 2017 20:25:41 +0100
Subject: [PATCH 2/3] Do not install all object files
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Jörg Krause <joerg.krause@embedded.rocks>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 61cd4210..b3c82b06 100644
--- a/Makefile
+++ b/Makefile
@@ -546,7 +546,7 @@ install-pkgconf : tt
 
 install-libs :
 	$(mkdir) $(installlibdir)
-	$(cp) $(objdir)* $(installlibdir)
+	$(cp) $(objdir)*.$(libext) $(installlibdir)
 
 install-includes :
 	$(mkdir) $(installincludedir)
-- 
2.12.0

