--- syslog-ng-3.29.1/lib/template/macros.c.orig	2020-08-10 14:04:39.000000000 -0400
+++ syslog-ng-3.29.1/lib/template/macros.c	2021-12-02 15:46:52.739340761 -0500
@@ -226,7 +226,7 @@ LogMacroDef macros[] =
 };
 
 
-static GTimeVal app_uptime;
+static gint64 app_uptime;
 static GHashTable *macro_hash;
 static LogTemplateOptions template_options_for_macro_expand;
 
@@ -679,10 +679,10 @@ log_macro_expand(GString *result, gint i
     }
     case M_SYSUPTIME:
     {
-      GTimeVal ct;
+      gint64 ct;
 
-      g_get_current_time(&ct);
-      format_uint64_padded(result, 0, 0, 10, g_time_val_diff(&ct, &app_uptime) / 1000 / 10);
+      ct = g_get_monotonic_time();
+      format_uint64_padded(result, 0, 0, 10, (ct - app_uptime) / 1000 / 10);
       break;
     }
 
@@ -720,7 +720,7 @@ log_macros_global_init(void)
   gint i;
 
   /* init the uptime (SYSUPTIME macro) */
-  g_get_current_time(&app_uptime);
+  app_uptime = g_get_monotonic_time();
   log_template_options_defaults(&template_options_for_macro_expand);
 
   macro_hash = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, NULL);
