diff -urN gst1-bcm-18.2-rdkv-20180727.orig/configure.ac gst1-bcm-18.2-rdkv-20180727/configure.ac
--- gst1-bcm-18.2-rdkv-20180727.orig/configure.ac	2018-12-17 20:32:10.000000000 +0000
+++ gst1-bcm-18.2-rdkv-20180727/configure.ac	2018-12-21 14:36:24.310768714 +0000
@@ -288,8 +288,6 @@
 
 plugindir="\$(libdir)/gstreamer-$GST_MAJORMINOR"
 PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 0.10.28])
-PKG_CHECK_MODULES([CURL], [libcurl >= 7.19.6])
-PKG_CHECK_MODULES([MPG123], [libmpg123 >= 1.19.0])
 AC_SUBST(PLUGIN_SRC)
 AC_SUBST(ENABLE_SVP)
 AC_SUBST(plugindir)
