diff --git a/modules/Alexa/build/BuildDefaults.cmake b/modules/Alexa/build/BuildDefaults.cmake
index 251b66e..9bdec43 100644
--- a/modules/Alexa/build/BuildDefaults.cmake
+++ b/modules/Alexa/build/BuildDefaults.cmake
@@ -23,7 +23,7 @@ macro(include_once module)
 endmacro()
 
 # Disallow out-of-source-builds.
-include_once(DisallowOutOfSourceBuilds)
+#include_once(DisallowOutOfSourceBuilds)
 
 # Setup default build options, like compiler flags and build type.
 include_once(BuildOptions)
