diff --git a/modules/Alexa/SampleApp/include/SampleApp/SampleApplication.h b/modules/Alexa/SampleApp/include/SampleApp/SampleApplication.h
index df998ac..6e430f9 100644
--- a/modules/Alexa/SampleApp/include/SampleApp/SampleApplication.h
+++ b/modules/Alexa/SampleApp/include/SampleApp/SampleApplication.h
@@ -154,7 +154,7 @@ public:
             alexaClientSDK::avsCommon::sdkInterfaces::ChannelVolumeInterface::Type speakerType);
     };
 
-private:
+protected:
     /**
      * Initialize a SampleApplication.
      *
