diff --git a/modules/Alexa/Utils/CMakeLists.txt b/modules/Alexa/Utils/CMakeLists.txt
index 7984dc3..79cc27d 100644
--- a/modules/Alexa/Utils/CMakeLists.txt
+++ b/modules/Alexa/Utils/CMakeLists.txt
@@ -8,3 +8,6 @@ add_library(Utils INTERFACE)
 
 target_include_directories(Utils INTERFACE
     "${Utils_SOURCE_DIR}/include")
+
+install(DIRECTORY "${Utils_SOURCE_DIR}/include/Utils" DESTINATION include
+    FILES_MATCHING PATTERN "*.h")
