diff -Nurpd a/Makefile b/Makefile
--- a/Makefile	2017-12-27 20:56:31.000000000 +0100
+++ b/Makefile	2019-11-07 09:52:49.554284682 +0100
@@ -176,7 +176,7 @@ install: $(RUN) libs
 	$(INSTALL) -d $(INST_LIBRARY)
 	$(INSTALL) $(LIBRARIES) $(INST_LIBRARY)
 ifdef LINKED_LIB_SH
-	cd $(INST_LIBRARY) && ln -s $(notdir $(LIB_SH) $(LINKED_LIB_SH))
+	cd $(INST_LIBRARY) && ln -sf $(notdir $(LIB_SH) $(LINKED_LIB_SH))
 endif
 	$(INSTALL) -d $(INST_BINARY)
 	$(INSTALL) $(RUN) $(INST_BINARY)
