diff -auNrd git.orig/Makefile.am git/Makefile.am
--- git.orig/Makefile.am	2018-01-30 18:08:28.906992629 -0500
+++ git/Makefile.am	2018-01-30 18:14:42.807230134 -0500
@@ -169,6 +169,7 @@
                             $(REFSW_VERSION)/common/event.c \
                             $(REFSW_VERSION)/common/fence_interface.c \
                             $(REFSW_VERSION)/common/fence_queue.c \
+                            $(REFSW_VERSION)/common/perf_event.cpp \
                             $(REFSW_VERSION)/common/queue.c \
                             $(REFSW_VERSION)/common/ring_buffer.c \
                             $(REFSW_VERSION)/common/surface_interface.c \
