diff -auNrd git.orig/Makefile.am git/Makefile.am
--- git.orig/Makefile.am	2018-06-13 15:39:38.763711188 -0400
+++ git/Makefile.am	2018-06-13 15:41:22.755979896 -0400
@@ -162,6 +162,7 @@
                             $(REFSW_VERSION)/nexus/display_nexus_multi.c \
                             $(REFSW_VERSION)/common/display_helpers.c \
                             $(REFSW_VERSION)/common/memory_convert.c \
+                            $(REFSW_VERSION)/common/memory_drm.c \
                             $(REFSW_VERSION)/common/memory_nexus.c \
                             $(REFSW_VERSION)/common/sched_nexus.c \
                             $(REFSW_VERSION)/common/display_framework.c \
@@ -185,6 +186,8 @@
                             $(REFSW_VERSION)/nexus/default_nexus.c \
                             $(REFSW_VERSION)/common/packet_yv12.c \
                             $(REFSW_VERSION)/common/packet_rgba.c \
+                            $(REFSW_VERSION)/common/perf_event.cpp \
+                            $(REFSW_VERSION)/common/nexus_surface_memory.c \
                             $(REFSW_VERSION)/common/memory_nexus.c \
                             $(REFSW_VERSION)/common/hardware_nexus.cpp \
                             $(REFSW_VERSION)/common/autoclif.c \
diff -auNrd git.orig/wayland-egl.cpp git/wayland-egl.cpp
--- git.orig/wayland-egl.cpp	2018-06-13 15:39:38.763711188 -0400
+++ git/wayland-egl.cpp	2018-06-13 15:39:21.415666527 -0400
@@ -1411,7 +1411,7 @@
       {
          void *pixmapHandle;
          NEXUS_SURFACEHANDLE surfaceHandle;
-         BEGL_PixmapInfo pixmapInfo;
+         BEGL_PixmapInfoEXT pixmapInfo;
 
          /*
           * Create a new Nexus surface/native pixmap pair
@@ -1423,7 +1423,7 @@
          #else
          pixmapInfo.format= BEGL_BufferFormat_eA8B8G8R8;
          #endif
-         if ( NXPL_CreateCompatiblePixmap(gNxplHandle, &pixmapHandle, &surfaceHandle, &pixmapInfo) )
+         if ( NXPL_CreateCompatiblePixmapEXT(gNxplHandle, &pixmapHandle, &surfaceHandle, &pixmapInfo) )
          {
             bpx->nativePixmap= pixmapHandle;
             bpx->surface= surfaceHandle;
