diff -auNrd git.orig/wayland-egl.cpp git/wayland-egl.cpp
--- git.orig/wayland-egl.cpp	2018-06-21 14:05:47.442325154 -0400
+++ git/wayland-egl.cpp	2018-06-21 14:05:19.558245662 -0400
@@ -844,6 +844,8 @@
       }
       printf("wayland-egl: eglGetDisplay: NxClient_Join ok as %s\n", joinSettings.name );
 
+      NxClient_UnregisterAcknowledgeStandby(NxClient_RegisterAcknowledgeStandby());
+
       NXPL_RegisterNexusDisplayPlatform( &gNxplHandle, 0 );
       if ( !gNxplHandle )
       {
