From 62260e0a42dd6c1c10660f68a531c49bd6cade2e Mon Sep 17 00:00:00 2001
From: Francois Wautier <francois@wautier.eu>
Date: Thu, 25 Feb 2021 20:07:08 +0700
Subject: [PATCH] Move README to reStructured text

Signed-off-by: Grzegorz Blach <grzegorz@blach.pl>
---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 582a994..315b3bb 100644
--- a/setup.py
+++ b/setup.py
@@ -4,7 +4,7 @@ import setuptools
 
 version = "0.2.7"
 
-with open("README.md", "r") as fh:
+with open("README.rst", "r") as fh:
     long_description = fh.read()
 
 setuptools.setup(
@@ -16,7 +16,7 @@ setuptools.setup(
     author_email="francois@wautier.eu",
     description="Scanning Bluetooth for advertised info with asyncio.",
     long_description=long_description,
-    long_description_content_type="text/markdown",
+    long_description_content_type="text/x-rst",
     url="http://github.com/frawau/aioblescan",
     keywords=["bluetooth", "advertising", "hci", "ble"],
     license="MIT",
-- 
2.27.0

