diff -ruN cobalt.orig/src/net/third_party/quic/core/frames/quic_frame.h cobalt/src/net/third_party/quic/core/frames/quic_frame.h
--- cobalt.orig/src/net/third_party/quic/core/frames/quic_frame.h	2021-09-07 17:58:28.083287161 +0530
+++ cobalt/src/net/third_party/quic/core/frames/quic_frame.h	2021-09-07 18:07:10.195682023 +0530
@@ -111,7 +111,7 @@
 static_assert(sizeof(QuicFrame) <= 24,
               "Frames larger than 24 bytes should be referenced by pointer.");
 #if !(defined(STARBOARD) && defined(__GNUC__) && !defined(__clang__) && \
-      __GNUC__ <= 7)
+      __GNUC__ <= 8)
 // Raspi compiler does not allow none-static type in offsetof.
 static_assert(offsetof(QuicStreamFrame, type) == offsetof(QuicFrame, type),
               "Offset of |type| must match in QuicFrame and QuicStreamFrame");
diff -ruN cobalt.orig/src/net/third_party/quic/core/frames/quic_inlined_frame.h cobalt/src/net/third_party/quic/core/frames/quic_inlined_frame.h
--- cobalt.orig/src/net/third_party/quic/core/frames/quic_inlined_frame.h	2021-09-07 17:58:28.083287161 +0530
+++ cobalt/src/net/third_party/quic/core/frames/quic_inlined_frame.h	2021-09-07 18:07:10.195682023 +0530
@@ -18,7 +18,7 @@
 struct QUIC_EXPORT_PRIVATE QuicInlinedFrame {
   QuicInlinedFrame(QuicFrameType type) : type(type) {
 #if !(defined(STARBOARD) && defined(__GNUC__) && !defined(__clang__) && \
-      __GNUC__ <= 7)
+      __GNUC__ <= 8)
     // Raspi compiler does not allow none-static type in offsetof, but we should
     // turn on this check on as many platforms as possible.
     static_assert(offsetof(DerivedT, type) == 0,
diff -ruN cobalt.orig/src/third_party/starboard/wpe/rpi/gyp_configuration.gypi cobalt/src/third_party/starboard/wpe/rpi/gyp_configuration.gypi
--- cobalt.orig/src/third_party/starboard/wpe/rpi/gyp_configuration.gypi	2021-09-07 18:01:28.002849795 +0530
+++ cobalt/src/third_party/starboard/wpe/rpi/gyp_configuration.gypi	2021-09-07 18:08:41.150350881 +0530
@@ -66,6 +66,11 @@
       # matters: Wall implies Wunused-parameter and Wno-unused-parameter
       # has no effect if specified before Wall.
       '-Wno-unused-parameter',
+      '-Wno-expansion-to-defined',
+      '-Wimplicit-fallthrough=0',
+      '-Wno-ignored-qualifiers',
+      '-Wno-error=class-memaccess',
+      '-Wno-error=deprecated-copy',
 
       # For some reason Thumb build is broken
       '-marm',
diff -ruN cobalt.orig/src/third_party/starboard/wpe/shared/player/player_internal.cc cobalt/src/third_party/starboard/wpe/shared/player/player_internal.cc
--- cobalt.orig/src/third_party/starboard/wpe/shared/player/player_internal.cc	2021-09-07 17:58:38.438241572 +0530
+++ cobalt/src/third_party/starboard/wpe/shared/player/player_internal.cc	2021-09-07 18:07:10.195682023 +0530
@@ -105,8 +105,7 @@
 
 G_END_DECLS
 
-#define GST_COBALT_SRC_GET_PRIVATE(obj) \
-  (G_TYPE_INSTANCE_GET_PRIVATE((obj), GST_COBALT_TYPE_SRC, GstCobaltSrcPrivate))
+#define GST_COBALT_SRC_GET_PRIVATE(obj) static_cast<GstCobaltSrcPrivate*>(gst_cobalt_src_get_instance_private(obj))
 
 struct _GstCobaltSrcPrivate {
   gchar* uri;
@@ -129,6 +128,7 @@
 G_DEFINE_TYPE_WITH_CODE(GstCobaltSrc,
                         gst_cobalt_src,
                         GST_TYPE_BIN,
+                        G_ADD_PRIVATE(GstCobaltSrc)
                         G_IMPLEMENT_INTERFACE(GST_TYPE_URI_HANDLER,
                                               gst_cobalt_src_uri_handler_init));
 
