--- systemd-245.5/shell-completion/zsh/_systemctl.in.orig	2020-05-05 11:17:35.011265230 -0400
+++ systemd-245.5/shell-completion/zsh/_systemctl.in	2020-05-05 11:18:31.941988715 -0400
@@ -221,7 +221,7 @@
     _systemctl_failed_units()  {_sys_failed_units=( ${${(f)"$(__systemctl list-units --state=failed "$PREFIX*" )"}%% *} ) }
 
 (( $+functions[_systemctl_unit_state] )) ||
-    _systemctl_unit_state() { typeset -gA _sys_unit_state; _sys_unit_state=( $(__systemctl list-unit-files "$PREFIX*" ) ) }
+    _systemctl_unit_state() { typeset -gA _sys_unit_state; _sys_unit_state=( $(__systemctl list-unit-files "$PREFIX*" | awk '{print $1, $2}') ) }
 
 local fun
 # Completion functions for ALL_UNITS
