From 5cf2d4e05673964e269e4b9a2e779d20f57c0d17 Mon Sep 17 00:00:00 2001
From: phunkyfish <phunkyfish@gmail.com>
Date: Sat, 6 Feb 2021 12:56:50 +0000
Subject: [PATCH] Remove register keyword as no longer valid in CPP17

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 src/lib/tsreader/DvbUtil.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/tsreader/DvbUtil.cpp b/src/lib/tsreader/DvbUtil.cpp
index 8c4798a..c973062 100644
--- a/src/lib/tsreader/DvbUtil.cpp
+++ b/src/lib/tsreader/DvbUtil.cpp
@@ -77,7 +77,7 @@ namespace MPTV
     //*******************************************************************
     uint32_t crc32(char *data, int len)
     {
-        register int i;
+        int i;
         uint32_t crc = 0xffffffff;
 
         for (i = 0; i < len; i++)
