diff --git a/examples/pxScene2d/src/pxScene2d.cpp b/examples/pxScene2d/src/pxScene2d.cpp
index e19a7d4..b9f6f42 100644
--- a/examples/pxScene2d/src/pxScene2d.cpp
+++ b/examples/pxScene2d/src/pxScene2d.cpp
@@ -2980,7 +2980,9 @@ void pxScriptView::runScript()
     mReady = new rtPromise();
 #endif
 
-    mCtx->runFile("init.js");
+    rtString s = getenv("SPARK_PATH");
+    s.append("init.js");
+    mCtx->runFile(s.cString());
 
     char buffer[MAX_URL_SIZE + 50];
     memset(buffer, 0, sizeof(buffer));
