diff --git a/examples/pxScene2d/src/pxTextBox.cpp b/examples/pxScene2d/src/pxTextBox.cpp
index 63243d4..84caef0 100644
--- a/examples/pxScene2d/src/pxTextBox.cpp
+++ b/examples/pxScene2d/src/pxTextBox.cpp
@@ -29,6 +29,7 @@ extern pxContext context;
 #include <math.h>
 #include <map>
 #include <stdlib.h>
+#include <cstring>
 
 static const char      isNewline_chars[] = "\n\v\f\r";
 static const char isWordBoundary_chars[] = " \t/:&,;.?!";
