diff --git a/examples/pxScene2d/external/dukluv/CMakeLists.txt b/examples/pxScene2d/external/dukluv/CMakeLists.txt
index 4afc8ab..abb52a7 100644
--- a/examples/pxScene2d/external/dukluv/CMakeLists.txt
+++ b/examples/pxScene2d/external/dukluv/CMakeLists.txt
@@ -8,7 +8,7 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/build)
 include(duktape.cmake)
 include(uv.cmake)
 
-add_library(duv
+add_library(duv STATIC
   src/duv.c
   src/refs.c
   src/utils.c
@@ -30,7 +30,7 @@ add_library(duv
 )
 
 
-add_library(dschema
+add_library(dschema STATIC
   src/schema.c
 )
 
