diff -ruN WPEBackend-rdk.orig/src/wpeframework/display.cpp WPEBackend-rdk/src/wpeframework/display.cpp
--- WPEBackend-rdk.orig/src/wpeframework/display.cpp	2020-04-28 14:31:45.981973616 +0530
+++ WPEBackend-rdk/src/wpeframework/display.cpp	2020-04-28 14:33:32.067995092 +0530
@@ -248,7 +248,7 @@
 
         g_source_add_poll(m_eventSource, &source->pfd);
         g_source_set_name(m_eventSource, "[WPE] Display");
-        g_source_set_priority(m_eventSource, G_PRIORITY_HIGH + 30);
+        g_source_set_priority(m_eventSource, G_PRIORITY_DEFAULT);
         g_source_set_can_recurse(m_eventSource, TRUE);
         g_source_attach(m_eventSource, g_main_context_get_thread_default());
     }
