--- linux/drivers/net/ethernet/broadcom/genet/bcmgenet.h	2015-04-25 06:30:00.000000000 +0200
+++ linux.hfgcc49/drivers/net/ethernet/broadcom/genet/bcmgenet.h	2015-11-03 23:41:28.760810385 +0100
@@ -21,7 +21,7 @@
 #define __BCMGENET_H__
 
 #define VERSION     "2.0"
-#define VER_STR     "v" VERSION " " __DATE__ " " __TIME__
+#define VER_STR     "v" VERSION
 
 #include <linux/skbuff.h>
 #include <linux/netdevice.h>
