From c0eab95389279db4212f1278eb436f37d62434ed Mon Sep 17 00:00:00 2001
From: bigbearishappy <953308023@qq.com>
Date: Fri, 3 Sep 2021 08:50:51 +0100
Subject: [PATCH] patch Makefile for buildroot compile

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index a4fb829..d3d4f3d 100644
--- a/Makefile
+++ b/Makefile
@@ -161,14 +161,14 @@ clean_%:
 	@for line in $(kmods); do make -C $(KBUILD) M=$(MOD_PATH)/$$line clean || exit; done
 
 install_%:
-	$(Q)$(MAKE) PLATFORM=$* install_arch
+	#$(Q)$(MAKE) PLATFORM=$* install_arch
 #	mkdir -p /lib/modules/$(uname_r)/extra/seeed || true
 	mkdir -p $(KO_DIR) || true
 	@for line in $(kmods); do \
 		echo $(MOD_PATH)/$$line/*.ko; \
 		cp $(MOD_PATH)/$$line/*.ko $(KO_DIR) || exit; \
 	done
-	@which depmod >/dev/null 2>&1 && depmod -a || true
+	#@which depmod >/dev/null 2>&1 && depmod -a || true
 
 ifeq ($(PLATFORM),)
 
-- 
2.20.1

