diff --git a/CMakeLists.txt b/CMakeLists.txt
index ec868cb..19e22ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,6 +60,8 @@ if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
     set(AAMP_CLI_LD_FLAGS "${AAMP_CLI_LD_FLAGS} -lgstvideo-1.0")
     string(STRIP ${AAMP_CLI_LD_FLAGS} AAMP_CLI_LD_FLAGS)
     set(AAMP_OS_SOURCES cocoa_window.mm)
+else()
+    set(AAMP_GST_VIDEO "-lgstvideo-1.0")
 endif(CMAKE_SYSTEM_NAME STREQUAL Darwin)
 
 find_package (Threads REQUIRED)
@@ -94,7 +96,7 @@ if(CMAKE_USE_RDK_PLUGINS)
 	endif()
 endif()
 
-set(LIBAAMP_DEPENDS ${OS_LD_FLAGS} ${GSTREAMERBASE_LIBRARIES} ${GSTREAMER_LIBRARIES} ${CURL_LIBRARIES} ${LIBDASH_LIBRARIES} ${LibXml2_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${OPENSSL_LIBRARIES} ${AAMP_CLI_LD_FLAGS} -ldl -labr -luuid ${SEC_CLIENT_LIB})
+set(LIBAAMP_DEPENDS ${OS_LD_FLAGS} ${GSTREAMERBASE_LIBRARIES} ${GSTREAMER_LIBRARIES} ${CURL_LIBRARIES} ${LIBDASH_LIBRARIES} ${LibXml2_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${OPENSSL_LIBRARIES} ${AAMP_CLI_LD_FLAGS} -ldl -labr -luuid ${SEC_CLIENT_LIB} ${AAMP_GST_VIDEO})
 set(PLAYBINTEST_DEPENDS ${OS_LD_FLAGS} ${CMAKE_THREAD_LIBS_INIT} ${GSTREAMER_LIBRARIES})
 
 if(CMAKE_QT5WEBKIT_JSBINDINGS)
diff --git a/aampgstplayer.cpp b/aampgstplayer.cpp
index a90dc62..93d02cf 100644
--- a/aampgstplayer.cpp
+++ b/aampgstplayer.cpp
@@ -33,6 +33,7 @@
 #include "priv_aamp.h"
 #include <pthread.h>
 #include <atomic>
+#include "gst/video/videooverlay.h"
 
 #ifdef AAMP_MPD_DRM
 #include "aampoutputprotection.h"
@@ -218,6 +219,12 @@ AAMPGstPlayer::~AAMPGstPlayer()
 	free(privateContext);
 }
 
+void AAMPGstPlayer::SetGLImageRectangle() {
+	int x, y, w, h;
+	sscanf(privateContext->videoRectangle, "%d,%d,%d,%d", &x, &y, &w, &h);
+	gst_video_overlay_set_render_rectangle(GST_VIDEO_OVERLAY(privateContext->video_sink), x, y, w, h);
+}
+
 /**
  * @brief Analyze stream info from the GstPipeline
  * @param[in] _this pointer to AAMPGstPlayer instance
@@ -632,6 +639,7 @@ bool AAMPGstPlayer_isVideoDecoder(const char* name, AAMPGstPlayer * _this)
 bool AAMPGstPlayer_isVideoSink(const char* name, AAMPGstPlayer * _this)
 {
 	return	(!_this->privateContext->using_westerossink && memcmp(name, "brcmvideosink", 13) == 0) || // brcmvideosink0, brcmvideosink1, ...
+			( !_this->privateContext->using_westerossink && memcmp(name, "glimagesinkbin0", 15) == 0) ||
 			( _this->privateContext->using_westerossink && memcmp(name, "westerossink", 12) == 0);
 }
 
@@ -894,18 +902,25 @@ static gboolean bus_message(GstBus * bus, GstMessage * msg, AAMPGstPlayer * _thi
 			{
 				if (AAMPGstPlayer_isVideoSink(GST_OBJECT_NAME(msg->src), _this))
 				{ // video scaling patch
+					_this->privateContext->video_sink = (GstElement *) msg->src;
+					if (memcmp(GST_OBJECT_NAME(msg->src), "glimagesinkbin0", 15) == 0)
+					{
+						_this->SetGLImageRectangle();
+					}
+					else
+					{
 					/*
 					brcmvideosink doesn't sets the rectangle property correct by default
 					gst-inspect-1.0 brcmvideosink
 					g_object_get(_this->privateContext->pipeline, "video-sink", &videoSink, NULL); - reports NULL
 					note: alternate "window-set" works as well
 					*/
-					_this->privateContext->video_sink = (GstElement *) msg->src;
-					logprintf("AAMPGstPlayer setting rectangle, video mute and zoom\n");
-					g_object_set(msg->src, "rectangle", _this->privateContext->videoRectangle, NULL);
-					g_object_set(msg->src, "zoom-mode", VIDEO_ZOOM_FULL == _this->privateContext->zoom ? 0 : 1, NULL);
-					g_object_set(msg->src, "show-video-window", !_this->privateContext->videoMuted, NULL);
-					g_object_set(msg->src, "enable-reject-preroll", FALSE, NULL);
+						logprintf("AAMPGstPlayer setting rectangle, video mute and zoom\n");
+						g_object_set(msg->src, "rectangle", _this->privateContext->videoRectangle, NULL);
+						g_object_set(msg->src, "zoom-mode", VIDEO_ZOOM_FULL == _this->privateContext->zoom ? 0 : 1, NULL);
+						g_object_set(msg->src, "show-video-window", !_this->privateContext->videoMuted, NULL);
+						g_object_set(msg->src, "enable-reject-preroll", FALSE, NULL);
+					}
 				}
 				else if (memcmp(GST_OBJECT_NAME(msg->src), "brcmaudiosink", 13) == 0)
 				{
@@ -2379,7 +2394,14 @@ void AAMPGstPlayer::SetVideoRectangle(int x, int y, int w, int h)
 #ifndef INTELCE
 	else if (privateContext->video_sink)
 	{
-		g_object_set(privateContext->video_sink, "rectangle", privateContext->videoRectangle, NULL);
+		if (memcmp(GST_OBJECT_NAME(privateContext->video_sink), "glimagesinkbin0", 15) == 0)
+		{
+			SetGLImageRectangle();
+		}
+		else
+		{
+			g_object_set(privateContext->video_sink, "rectangle", privateContext->videoRectangle, NULL);
+		}
 	}
 #else
 #if defined(INTELCE_USE_VIDRENDSINK)
diff --git a/aampgstplayer.h b/aampgstplayer.h
index 9edb841..bd3ce0f 100644
--- a/aampgstplayer.h
+++ b/aampgstplayer.h
@@ -76,6 +76,8 @@ public:
 	void NotifyFirstFrame(MediaType type);
 	void DumpDiagnostics();
 	void SignalTrickModeDiscontinuity();
+
+	void SetGLImageRectangle();
 private:
 	void PauseAndFlush(bool playAfterFlush);
 	void TearDownStream(MediaType mediaType);
