From 1abd7714e2fd47033fc24e064dba969b1f14e726 Mon Sep 17 00:00:00 2001
From: Srinivas Kakarla <srinivas.kakarla@valuelabs.com>
Date: Thu, 14 Jul 2016 17:12:51 +0530
Subject: [PATCH] Fix to set current_fragment for live streaming.

---
 ext/smoothstreaming/gstmssmanifest.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/ext/smoothstreaming/gstmssmanifest.c b/ext/smoothstreaming/gstmssmanifest.c
index 55331a9..8fca8b1 100644
--- a/ext/smoothstreaming/gstmssmanifest.c
+++ b/ext/smoothstreaming/gstmssmanifest.c
@@ -1211,6 +1211,11 @@ gst_mss_stream_seek (GstMssStream * stream, gboolean forward,
   GST_DEBUG ("Stream %s seeking to %" G_GUINT64_FORMAT, stream->url, time);
   for (iter = stream->fragments; iter; iter = g_list_next (iter)) {
     fragment = iter->data;
+    if(stream->has_live_fragments){
+      if (fragment->time + fragment->repetitions * fragment->duration > time)
+        stream->current_fragment = iter;
+      break;
+    }
     if (fragment->time + fragment->repetitions * fragment->duration > time) {
       stream->current_fragment = iter;
       stream->fragment_repetition_index =
@@ -1314,7 +1319,8 @@ gst_mss_stream_reload_fragments (GstMssStream * stream, xmlNodePtr streamIndex)
   if (builder.fragments) {
     g_list_free_full (stream->fragments, g_free);
     stream->fragments = g_list_reverse (builder.fragments);
-    stream->current_fragment = stream->fragments;
+    if (!stream->has_live_fragments)
+      stream->current_fragment = stream->fragments;
     /* TODO Verify how repositioning here works for reverse
      * playback - it might start from the wrong fragment */
     gst_mss_stream_seek (stream, TRUE, 0, current_gst_time, NULL);
-- 
1.9.1

