diff --git a/ext/smoothstreaming/gstmssmanifest.c b/ext/smoothstreaming/gstmssmanifest.c
index 291080a..480114d 100644
--- a/ext/smoothstreaming/gstmssmanifest.c
+++ b/ext/smoothstreaming/gstmssmanifest.c
@@ -1465,10 +1465,14 @@ gst_mss_manifest_change_bitrate (GstMssManifest * manifest, guint64 bitrate)
    * it should actually use the sum of all streams bitrates to compare to
    * the target value */
 
-  if (bitrate == 0) {
-    /* use maximum */
-    bitrate = G_MAXUINT64;
-  }
+  /* TODO For now, we will comment out this part of the code. We want to use
+   * the lowest quality of the stream to start playback. With this code,
+   * it starts with the highest.
+   */
+//  if (bitrate == 0) {
+//    /* use maximum */
+//    bitrate = G_MAXUINT64;
+//  }
 
   for (iter = gst_mss_manifest_get_streams (manifest); iter;
       iter = g_slist_next (iter)) {
