Install includes like lua's ones

Signed-off-by: Francois Perrad <francois.perrad@gadz.org>

Index: b/Makefile
===================================================================
--- a/Makefile
+++ b/Makefile
@@ -39,7 +39,7 @@
 INSTALL_LIB=   $(DPREFIX)/$(MULTILIB)
 INSTALL_SHARE_= $(PREFIX)/share
 INSTALL_SHARE= $(DESTDIR)$(INSTALL_SHARE_)
-INSTALL_DEFINC= $(DPREFIX)/include/luajit-$(MMVERSION)
+INSTALL_DEFINC= $(DPREFIX)/include
 INSTALL_INC=   $(INSTALL_DEFINC)
 
 export INSTALL_LJLIBD= $(INSTALL_SHARE_)/luajit-$(MMVERSION)
