diff --git a/westeros-nested.cpp b/westeros-nested.cpp
index b241a48..a36efe2 100644
--- a/westeros-nested.cpp
+++ b/westeros-nested.cpp
@@ -1040,7 +1040,7 @@ void WstNestedConnectionAttachAndCommitDevice( WstNestedConnection *nc,
       struct wl_buffer *buffer;
       
       buffer= wl_sb_create_buffer( nc->sb, 
-                                   (uint32_t)deviceBuffer, 
+                                   (uintptr_t)deviceBuffer, 
                                    width, 
                                    height, 
                                    stride,
