diff --git a/brcm/Makefile.am b/brcm/Makefile.am
index 2bfc998..725a59d 100644
--- a/brcm/Makefile.am
+++ b/brcm/Makefile.am
@@ -46,7 +46,13 @@ libwesteros_gl_la_CXXFLAGS= $(AM_CFLAGS)
 libwesteros_gl_la_LDFLAGS= \
    $(AM_LDFLAGS) \
    -lwayland-egl \
-   -lnexus
+   ${NEXUS_LD_LIBRARIES}
+
+if NXCLIENT_LOCAL
+libwesteros_gl_la_LDFLAGS += -lnxclient_local
+else
+libwesteros_gl_la_LDFLAGS += -lnxclient
+endif
 
 distcleancheck_listfiles = *-libtool
 
diff --git a/brcm/configure.ac b/brcm/configure.ac
index f18567f..c9dea91 100644
--- a/brcm/configure.ac
+++ b/brcm/configure.ac
@@ -44,6 +44,20 @@ AC_HEADER_STDBOOL
 
 IARM_CFLAGS=" "
 
+NXCLIENT_LOCAL=" "
+AC_ARG_ENABLE([nxclient_local],
+        AS_HELP_STRING([--enable-nxclient_local],[enable local nxclient ]),
+        [
+          case "${enableval}" in
+           yes) NXCLIENT_LOCAL=true ;;
+           no) NXCLIENT_LOCAL=false ;;
+          *) AC_MSG_ERROR([bad value ${enableval} for --enable-nxclient_local ]) ;;
+           esac
+           ],
+         [echo "local nxclient is enable"])
+AM_CONDITIONAL([NXCLIENT_LOCAL], [test x$NXCLIENT_LOCAL = xtrue])
+AC_SUBST(NXCLIENT_LOCAL)
+
 # Checks for library functions.
 #Add the subdirectories to be considered for building.
 SUBDIRS=" "
diff --git a/brcm/westeros-render-nexus/Makefile.am b/brcm/westeros-render-nexus/Makefile.am
index 6f8f762..31dc514 100644
--- a/brcm/westeros-render-nexus/Makefile.am
+++ b/brcm/westeros-render-nexus/Makefile.am
@@ -30,12 +30,18 @@ lib_LTLIBRARIES = libwesteros_render_nexus.la
 
 libwesteros_render_nexus_la_SOURCES = westeros-render-nexus.cpp
 libwesteros_render_nexus_la_includedir = $(includedir)
-libwesteros_render_nexus_la_LDFLAGS= -lnxclient -lwesteros_simplebuffer_server
+libwesteros_render_nexus_la_LDFLAGS= -lwesteros_simplebuffer_server
 if HAVE_WAYLAND_EGL
 AM_CXXFLAGS += -DWESTEROS_HAVE_WAYLAND_EGL
 libwesteros_render_nexus_la_LDFLAGS += -lwayland-egl
 endif
 
+if NXCLIENT_LOCAL
+libwesteros_render_nexus_la_LDFLAGS += -lnxclient_local
+else
+libwesteros_render_nexus_la_LDFLAGS += -lnxclient
+endif
+
 distcleancheck_listfiles = *-libtool
 
 ## IPK Generation Support
diff --git a/brcm/westeros-render-nexus/configure.ac b/brcm/westeros-render-nexus/configure.ac
index a706f87..0d690f8 100644
--- a/brcm/westeros-render-nexus/configure.ac
+++ b/brcm/westeros-render-nexus/configure.ac
@@ -45,6 +45,19 @@ AC_HEADER_STDBOOL
 WAYLAND_EGL_DETECTED=" "
 
 IARM_CFLAGS=" "
+NXCLIENT_LOCAL=" "
+AC_ARG_ENABLE([nxclient_local],
+        AS_HELP_STRING([--enable-nxclient_local],[enable local nxclient ]),
+        [
+          case "${enableval}" in
+           yes) NXCLIENT_LOCAL=true ;;
+           no) NXCLIENT_LOCAL=false ;;
+          *) AC_MSG_ERROR([bad value ${enableval} for --enable-nxclient_local ]) ;;
+           esac
+           ],
+         [echo "local nxclient is enable"])
+AM_CONDITIONAL([NXCLIENT_LOCAL], [test x$NXCLIENT_LOCAL = xtrue])
+AC_SUBST(NXCLIENT_LOCAL)
 
 # Checks for library functions.
 #Add the subdirectories to be considered for building.
