diff --git a/brcm/westeros-sink/westeros-sink-soc.c b/brcm/westeros-sink/westeros-sink-soc.c
index 8cb1102..bf0f765 100644
--- a/brcm/westeros-sink/westeros-sink-soc.c
+++ b/brcm/westeros-sink/westeros-sink-soc.c
@@ -2224,8 +2224,8 @@ void gst_westeros_sink_soc_update_video_position( GstWesterosSink *sink )
             vComposition.virtualDisplay.height= 480;
             break;
          default:
-            vComposition.virtualDisplay.width= 1280;
-            vComposition.virtualDisplay.height= 720;
+            vComposition.virtualDisplay.width= 1920;
+            vComposition.virtualDisplay.height= 1080;
             break;
       }
 
