diff -auNrd git.orig/brcm/westeros-sink/westeros-sink-soc.c git/brcm/westeros-sink/westeros-sink-soc.c
--- git.orig/brcm/westeros-sink/westeros-sink-soc.c	2018-01-10 17:11:09.000465436 -0500
+++ git/brcm/westeros-sink/westeros-sink-soc.c	2018-01-10 17:09:04.639753223 -0500
@@ -1477,8 +1477,8 @@
             vClientSettings.composition.virtualDisplay.height= 480;
             break;
          default:
-            vClientSettings.composition.virtualDisplay.width= 1280;
-            vClientSettings.composition.virtualDisplay.height= 720;
+            vClientSettings.composition.virtualDisplay.width= 1920;
+            vClientSettings.composition.virtualDisplay.height= 1080;
             break;
       }
 
