commit 90f094bee4dd8ba1b5a99cb6bead3a70602c56e9
Author: HaseenaSainul <h.sainul@metrological.com>
Date:   Thu May 7 20:25:59 2020 +0530

    Westeros-sink:brcm: video scaling based on platform support

diff --git a/brcm/westeros-sink/westeros-sink-soc.c b/brcm/westeros-sink/westeros-sink-soc.c
index 7cb8f1e..f3c6be7 100644
--- a/brcm/westeros-sink/westeros-sink-soc.c
+++ b/brcm/westeros-sink/westeros-sink-soc.c
@@ -336,6 +336,8 @@ static bool isSVPEnabled( void )
    bool enabled= false;
    #if NEXUS_PLATFORM_VERSION_MAJOR >= 16
    NEXUS_VideoDecoderCapabilities videoDecoderCap;
+   memset(&videoDecoderCap, 0, sizeof(NEXUS_VideoDecoderCapabilities));
+
    NEXUS_GetVideoDecoderCapabilities(&videoDecoderCap);
    enabled=  (videoDecoderCap.memory[0].secure == NEXUS_SecureVideo_eSecure) ? true : false;
    #endif
@@ -357,7 +359,7 @@ static bool useSecureGraphics( void )
    return useSecure;
 }
 
-void resourceChangedCallback( void *context, int param )
+static void resourceChangedCallback( void *context, int param )
 {
    NEXUS_Error rc;
    GstWesterosSink *sink= (GstWesterosSink*)context;
@@ -387,6 +389,11 @@ static void streamChangedCallback(void * context, int param)
 
    NEXUS_SimpleVideoDecoder_GetStreamInformation(decoderHandle, &streamInfo);
    #if (NEXUS_PLATFORM_VERSION_MAJOR > 17) || ((NEXUS_PLATFORM_VERSION_MAJOR == 17) && (NEXUS_PLATFORM_VERSION_MINOR > 1))
+   #ifdef ENABLE_DOLBYVISION
+   NxClient_DisplaySettings displaySettings;
+   NEXUS_Error rc= NEXUS_SUCCESS;
+   NEXUS_VideoDynamicRangeMode dynamicRangeMode= NEXUS_VideoDynamicRangeMode_eTrackInput;
+   #endif
    switch (streamInfo.dynamicMetadataType)
    {
       case NEXUS_VideoDecoderDynamicRangeMetadataType_eDolbyVision:
@@ -2224,8 +2231,29 @@ void gst_westeros_sink_soc_update_video_position( GstWesterosSink *sink )
             vComposition.virtualDisplay.height= 480;
             break;
          default:
-            vComposition.virtualDisplay.width= 1920;
-            vComposition.virtualDisplay.height= 1080;
+            {
+               NEXUS_SurfaceClientStatus scStatus;
+               bool using1080Graphics= false;
+               int rc= NEXUS_SurfaceClient_GetStatus( sink->soc.surfaceClient, &scStatus );
+               if ( rc == NEXUS_SUCCESS )
+               {
+                  if ( (scStatus.display.framebuffer.width == 1920) &&
+                       (scStatus.display.framebuffer.height == 1080) )
+                  {
+                     using1080Graphics= true;
+                  }
+               }
+               if ( using1080Graphics )
+               {
+                  vComposition.virtualDisplay.width= 1920;
+                  vComposition.virtualDisplay.height= 1080;
+               }
+               else
+               {
+                  vComposition.virtualDisplay.width= 1280;
+                  vComposition.virtualDisplay.height= 720;
+               }
+            }
             break;
       }
 
