diff --git a/netflix/src/base/system/Endian.h b/netflix/src/base/system/Endian.h
index bb4d970..6a5e3ac 100644
--- a/netflix/src/base/system/Endian.h
+++ b/netflix/src/base/system/Endian.h
@@ -24,7 +24,7 @@
 // ing file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 #if !defined(NRDP_LITTLE_ENDIAN) && !defined(NRDP_BIG_ENDIAN) && !defined(NRDP_PDP_ENDIAN)
-#if defined (__GLIBC__)
+#if !defined (__UCLIBC__)
 #  include <endian.h>
 #if (__BYTE_ORDER == __LITTLE_ENDIAN)
 #   define NRDP_LITTLE_ENDIAN
diff --git a/netflix/src/platform/gibbon/GibbonConsole.cpp b/netflix/src/platform/gibbon/GibbonConsole.cpp
index 10306d9..7ea8f38 100644
--- a/netflix/src/platform/gibbon/GibbonConsole.cpp
+++ b/netflix/src/platform/gibbon/GibbonConsole.cpp
@@ -71,7 +71,7 @@ extern "C" int tgetnum(char *);
 #include <unistd.h>
 
 
-#if defined(NRDP_PLATFORM_LINUX) && !defined(__APPLE__) && !defined(__UCLIBC__)
+#if defined(NRDP_PLATFORM_LINUX) && !defined(__APPLE__) && defined(__GLIBC__)
 #include <gnu/libc-version.h>
 #endif
 
diff --git a/partner/dpi/reference/FileSystem.cpp b/partner/dpi/reference/FileSystem.cpp
index ec8a926..5eea6aa 100644
--- a/partner/dpi/reference/FileSystem.cpp
+++ b/partner/dpi/reference/FileSystem.cpp
@@ -28,6 +28,8 @@
 #include <sys/mman.h>
 #endif
 
+#include <dlfcn.h>
+
 #include <nrdbase/Assert.h>
 #include <nrdbase/Base64.h>
 #include <nrdbase/OpenSSLLib.h>
