From 09d4632828fc7f1a51da8673adb01b233fde61da Mon Sep 17 00:00:00 2001
From: Bram Oosterhuis <bram.oosterhuis.nl@gmail.com>
Date: Fri, 12 Aug 2016 19:53:03 +0200
Subject: [PATCH] [input] mapped linux media contol keys.

---
 partner/input/devinput/InputManagerDevInput.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/partner/input/devinput/InputManagerDevInput.cpp b/partner/input/devinput/InputManagerDevInput.cpp
index 383f073..779971a 100644
--- a/partner/input/devinput/InputManagerDevInput.cpp
+++ b/partner/input/devinput/InputManagerDevInput.cpp
@@ -81,6 +81,10 @@ static inline KeyEvent::Key keyCodeToKey(uint16_t c)
     case KEY_DELETE: code = KeyEvent::GIBBON_KEY_DELETE; break;
     case KEY_PRINT: code = KeyEvent::GIBBON_KEY_PRINT; break;
     case KEY_PAUSE: code = KeyEvent::GIBBON_KEY_PAUSE; break;
+    case KEY_REWIND: code = KeyEvent::GIBBON_KEY_F7; break;
+    case KEY_PLAYPAUSE: code = KeyEvent::GIBBON_KEY_F8; break;
+    case KEY_FORWARD: code = KeyEvent::GIBBON_KEY_F9; break;
+    case KEY_STOPCD: code = KeyEvent::GIBBON_KEY_F10; break;
     default:
         break;
     }
