--- apparmor-3.0.3/libraries/libapparmor/swig/python/Makefile.am.orig	2021-10-14 11:45:19.652627942 -0400
+++ apparmor-3.0.3/libraries/libapparmor/swig/python/Makefile.am	2021-10-14 11:46:14.010656964 -0400
@@ -14,7 +14,7 @@
 
 all-local: libapparmor_wrap.c setup.py
 	if test ! -f libapparmor_wrap.c; then cp $(srcdir)/libapparmor_wrap.c . ; fi
-	CC="$(CC)" CFLAGS="$(PYTHON_CPPFLAGS) $(EXTRA_WARNINGS)" LDSHARED="$(CC) -shared" LDFLAGS="$(PYTHON_LDFLAGS) $(LDFLAGS)" $(PYTHON) setup.py build
+	$(PYTHON_DISTUTILS_OPTS) CC="$(CC)" CFLAGS="$(PYTHON_CPPFLAGS) $(EXTRA_WARNINGS)" LDSHARED="$(CC) -shared" LDFLAGS="$(PYTHON_LDFLAGS) $(LDFLAGS)" $(PYTHON) setup.py build
 
 install-exec-local:
 	$(PYTHON) setup.py install --root="/$(DESTDIR)" --prefix="$(prefix)"
