From f213fd29c1015a3afee9a3fde0dd7ce6c6325802 Mon Sep 17 00:00:00 2001
From: Ricardo Martincoski <ricardo.martincoski@gmail.com>
Date: Fri, 27 May 2022 20:15:00 -0300
Subject: [PATCH 1/1] package/binutils: change .mk

Signed-off-by: Ricardo Martincoski <ricardo.martincoski@gmail.com>
---
 package/binutils/binutils.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/binutils/binutils.mk b/package/binutils/binutils.mk
index 8c4a1371ca..3f143d9d76 100644
--- a/package/binutils/binutils.mk
+++ b/package/binutils/binutils.mk
@@ -1,6 +1,6 @@
 ################################################################################
 #
-# binutils
+# Binutils
 #
 ################################################################################
--
2.25.1
