--- optee-test-3.4.0/CMakeLists.txt.orig	2020-03-30 07:27:22.057733993 -0400
+++ optee-test-3.4.0/CMakeLists.txt	2020-03-30 07:27:55.403006728 -0400
@@ -18,7 +18,7 @@
 	-Wmissing-noreturn -Wmissing-prototypes -Wnested-externs
 	-Wpointer-arith -Wshadow -Wstrict-prototypes
 	-Wswitch-default -Wunsafe-loop-optimizations
-	-Wwrite-strings -Werror -fPIC
+	-Wwrite-strings -Wno-format-overflow -Werror -fPIC
  	-Wno-missing-field-initializers
 )
 
