--- playready/src/CMakeLists.txt	2015-10-13 13:15:12.437497335 +0200
+++ playready/src/CMakeLists.txt	2015-10-13 14:48:14.770201737 +0200
@@ -49,9 +49,9 @@
 include( "sources.cmake" )
 
 #libplayready!
-set (DEST_INC "include/playready2.5-ss-tee")
-add_library( playready-2.5-ss-tee ${CPP_SRCS} )
-install( TARGETS playready-2.5-ss-tee DESTINATION lib )
+set (DEST_INC "include/playready")
+add_library( playready ${CPP_SRCS} )
+install( TARGETS playready DESTINATION lib )
 install( FILES "playready.cmake"  DESTINATION ${DEST_INC} )
 install( DIRECTORY inc/           DESTINATION ${DEST_INC}     FILES_MATCHING PATTERN "*.h" )
 install( DIRECTORY oem/           DESTINATION ${DEST_INC}/oem FILES_MATCHING PATTERN "*.h" )
--- playready/src/sources.cmake	2015-10-15 10:35:56.478168917 +0200
+++ playready/src/sources.cmake	2015-10-15 10:37:41.988279921 +0200
@@ -423,7 +423,7 @@
     oem/common/ecc/base/oemeccp256.c
     oem/common/hal/CoreAPI/oemderivedkeys.c
     oem/common/hal/CoreAPI/oemhal.c
-#    oem/common/hal/CoreAPI/oemhalree.c
+    oem/common/hal/CoreAPI/oemhalree.c
     oem/common/hal/CoreAPI/oemhaldevcerts.c
     oem/common/hal/CoreAPI/oemsparsexmrparser.c
     oem/common/hal/oemimpl/oemhaloemimpl.c
@@ -476,7 +476,7 @@
         oem/netflix/oemimplrand.c
         oem/netflix/oeminfo.c
         oem/netflix/oemtime.c
-        oem/common/hal/CoreAPI/oemhalree.c
+#        oem/common/hal/CoreAPI/oemhalree.c
     )
 else()
     set(CPP_SRCS ${CPP_SRCS}

