diff --git a/psplash-colors.h b/psplash-colors.h
index 82a9893..c681d81 100644
--- a/psplash-colors.h
+++ b/psplash-colors.h
@@ -12,10 +12,10 @@
 #define _HAVE_PSPLASH_COLORS_H
 
 /* This is the overall background color */
-#define PSPLASH_BACKGROUND_COLOR 0xec,0xec,0xe1
+#define PSPLASH_BACKGROUND_COLOR 0x00,0x00,0x00
 
 /* This is the color of any text output */
-#define PSPLASH_TEXT_COLOR 0x6d,0x6d,0x70
+#define PSPLASH_TEXT_COLOR 0xff,0xff,0xff
 
 /* This is the color of the progress bar indicator */
 #define PSPLASH_BAR_COLOR 0x6d,0x6d,0x70
diff --git a/psplash-config.h b/psplash-config.h
index 0ba8440..78ddff3 100644
--- a/psplash-config.h
+++ b/psplash-config.h
@@ -22,9 +22,9 @@
 #endif
 
 /* Position of the image split from top edge, numerator of fraction */
-#define PSPLASH_IMG_SPLIT_NUMERATOR 5
+#define PSPLASH_IMG_SPLIT_NUMERATOR 2
 
 /* Position of the image split from top edge, denominator of fraction */
-#define PSPLASH_IMG_SPLIT_DENOMINATOR 6
+#define PSPLASH_IMG_SPLIT_DENOMINATOR 3
 
 #endif
