From e29c27fafd991644d3dba2db45bfca02f897ce8b Mon Sep 17 00:00:00 2001
From: Wouter Meek <woutermeek@users.noreply.github.com>
Date: Tue, 9 Aug 2016 13:35:42 +0200
Subject: [PATCH] [QueuePlayer] do not flush Queue after stop

---
 QueuePlayer/CInterface.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/QueuePlayer/CInterface.cpp b/QueuePlayer/CInterface.cpp
index 59cbcf7..cb2b2bc 100644
--- a/QueuePlayer/CInterface.cpp
+++ b/QueuePlayer/CInterface.cpp
@@ -1332,7 +1332,7 @@ class QueuePlayerWrapper : public Solutions::QueuePlayer::INotifications {
             _timer.Revoke(_pendingTransition);
             _timer.Revoke(_abortTransition);
             _controller.Stop();
-            _controller.Flush();
+            //_controller.Flush();
         }
  
         if (_buffer != nullptr) {
