From fb49b6a0ce9bc0d8d8a7577dd7fa09bdfb275acc Mon Sep 17 00:00:00 2001
From: Wouter Meek <woutermeek@gmail.com>
Date: Tue, 9 Aug 2016 05:21:13 -0700
Subject: [PATCH] [QueuePlayer] log stream type

---
 QueuePlayer/CInterface.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/QueuePlayer/CInterface.cpp b/QueuePlayer/CInterface.cpp
index cb2b2bc..7e1b468 100644
--- a/QueuePlayer/CInterface.cpp
+++ b/QueuePlayer/CInterface.cpp
@@ -320,7 +320,7 @@ class QueuePlayerWrapper : public Solutions::QueuePlayer::INotifications {
 
                 if(counter < 20) {
                     counter++;
-                    printf ("pts32bit: %d, pts :%llu  ,dts: %llu, %s _offsetDts: %llu, _offsetPts: %llu \n", metaDataBuffer[2], _lastPts, _lastDts, IsTypeAudio() ? "AUDIO": "VIDEO", _staticDtsOffset, _staticPtsOffset);
+                    printf ("pts32bit: %d, pts :%llu  ,dts: %llu, %s _offsetDts: %llu, _offsetPts: %llu, _type: %d \n", metaDataBuffer[2], _lastPts, _lastDts, IsTypeAudio() ? "AUDIO": "VIDEO", _staticDtsOffset, _staticPtsOffset, _type);
                 }
 
                 memset ( &(metaDataBuffer[11]), 0, 24);
