diff -purN python-connexion-2.14.0.orig/setup.py python-connexion-2.14.0/setup.py
--- python-connexion-2.14.0.orig/setup.py	2022-06-22 12:23:43.000000000 -0700
+++ python-connexion-2.14.0/setup.py	2022-08-18 13:38:35.122259935 -0700
@@ -100,7 +100,7 @@ setup(
     url='https://github.com/zalando/connexion',
     keywords='openapi oai swagger rest api oauth flask microservice framework',
     license='Apache License Version 2.0',
-    setup_requires=['flake8'],
+    setup_requires=[],
     python_requires=">=3.6",
     install_requires=install_requires + flask_require,
     tests_require=tests_require,
