diff -ruN WPEWebKit.orig/Source/JavaScriptCore/runtime/GenericTypedArrayView.h WPEWebKit/Source/JavaScriptCore/runtime/GenericTypedArrayView.h
--- WPEWebKit.orig/Source/JavaScriptCore/runtime/GenericTypedArrayView.h	2020-08-26 16:01:55.000000000 +0530
+++ WPEWebKit/Source/JavaScriptCore/runtime/GenericTypedArrayView.h	2021-01-10 20:16:29.323386019 +0530
@@ -27,6 +27,7 @@
 
 #include "ArrayBuffer.h"
 #include "ArrayBufferView.h"
+#include "HeapInlines.h"
 
 namespace JSC {
 
