From cbdfde1abc7715f7c113e3c2144953d1d25cf8db Mon Sep 17 00:00:00 2001
From: Vanya Sergeev <vsergeev@gmail.com>
Date: Wed, 23 Dec 2015 11:30:33 +0100
Subject: [PATCH] Disable buggy_getaddrinfo configure test when cross-compiling
 with IPv6 support

Signed-off-by: Vanya Sergeev <vsergeev@gmail.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 879e6a2d432..7bc049863c3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5433,7 +5433,7 @@ fi]))
 dnl if ac_cv_func_getaddrinfo
 ])
 
-if test "$ac_cv_func_getaddrinfo" = no -o "$ac_cv_buggy_getaddrinfo" = yes
+if test "$ac_cv_func_getaddrinfo" = no || test "$cross_compiling" != "yes" -a "$ac_cv_buggy_getaddrinfo" = yes
 then
   AS_VAR_IF([ipv6], [yes], [
     AC_MSG_ERROR([m4_normalize([
-- 
2.39.5

