From 17d7b6d56cf554dcc6fcb263e1fd2b9e3dbf4320 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Wed, 23 Dec 2015 11:51:58 +0100
Subject: [PATCH] Add an option to disable the ossaudiodev module

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 configure.ac | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/configure.ac b/configure.ac
index 20b3c2459f4..87853f47329 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3728,6 +3728,12 @@ fi
 
 AC_MSG_RESULT($with_decimal_contextvar)
 
+AC_ARG_ENABLE(ossaudiodev,
+	AS_HELP_STRING([--disable-ossaudiodev], [disable OSSAUDIODEV]),
+	[ if test "$enableval" = "no"; then
+	  DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} ossaudiodev"
+	  fi])
+
 # Check for libmpdec machine flavor
 AC_MSG_CHECKING(for decimal libmpdec machine)
 AS_CASE([$ac_sys_system],
-- 
2.44.0

