From 2f08acf85424eabff501e26b01f5854e2aa4a5ce Mon Sep 17 00:00:00 2001
From: Avery Pennarun <apenwarr@gmail.com>
Date: Mon, 11 Jul 2016 20:32:23 -0400
Subject: [PATCH] remove xattrs

---
 toys.h          | 2 +-
 toys/posix/cp.c | 5 ++++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/toys.h b/toys.h
index 414f439..ff88c87 100644
--- a/toys.h
+++ b/toys.h
@@ -68,7 +68,7 @@
 #include <sys/sysinfo.h>
 
 #include "lib/lib.h"
-#include "lib/lsm.h"
+//#include "lib/lsm.h"
 #include "lib/toyflags.h"
 #include "toys/e2fs.h"
 
diff --git a/toys/posix/cp.c b/toys/posix/cp.c
index bedcfaa..782d677 100644
--- a/toys/posix/cp.c
+++ b/toys/posix/cp.c
@@ -118,9 +118,12 @@ GLOBALS(
 struct cp_preserve {
   char *name;
 } static const cp_preserve[] = TAGGED_ARRAY(CP,
-  {"mode"}, {"ownership"}, {"timestamps"}, {"context"}, {"xattr"},
+  {"mode"}, {"ownership"}, {"timestamps"},
 );
 
+#define _CP_xattr 0
+#define _CP_context 0
+
 // Callback from dirtree_read() for each file/directory under a source dir.
 
 int cp_node(struct dirtree *try)
-- 
2.8.0.rc3.226.g39d4020

