diff --git a/CMakeLists.txt b/CMakeLists.txt
index ea4a424..cbcdc0a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,7 +96,9 @@ add_library(gstaamp SHARED ${GSTAAMP_SOURCES})
 
 if(CMAKE_DASH_DRM)
 	target_include_directories (gstaamp PRIVATE drm/gst)
-	set(AAMP_COMMON_DEPENDENCIES "${AAMP_COMMON_DEPENDENCIES} -lIARMBus -lds -lsystemd")
+	if(CMAKE_RDK_VIDEO_BUILD)
+		set(AAMP_COMMON_DEPENDENCIES "${AAMP_COMMON_DEPENDENCIES} -lIARMBus -lds -lsystemd")
+	endif()
 	if(CMAKE_USE_OPENCDM)
 		message("CMAKE_USE_OPENCDM set")
 		if (DEFINED SECAPI_ENGINE_BROADCOM_SAGE)
@@ -118,6 +120,9 @@ if(CMAKE_DASH_DRM)
 		endif()
 		find_path (STAGING_INCDIR opencdm)
 		include_directories(${STAGING_INCDIR}/opencdm)
+	elseif(CMAKE_USE_OPENCDM_ADAPTER)
+		set(AAMP_COMMON_DEPENDENCIES "${AAMP_COMMON_DEPENDENCIES}  -locdm")
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_OPENCDM")
 	else()
 		message("CMAKE_USE_OPENCDM not set")
 		if (DEFINED SECAPI_ENGINE_BROADCOM_SAGE)
