diff --git a/src/bcm-nexus/CMakeLists.txt b/src/bcm-nexus/CMakeLists.txt
index 57ec6ff..5dd8916 100644
--- a/src/bcm-nexus/CMakeLists.txt
+++ b/src/bcm-nexus/CMakeLists.txt
@@ -12,6 +12,15 @@ if (EGL_LIBRARIES_nxclient)
     list(APPEND WPE_PLATFORM_LIBRARIES nxclient)
 endif()
 
+# TODO: write a pc file for this.
+if (NEXUS_CLIENT_MODE)
+    message(STATUS "Building with nexus in client mode" )
+    list(APPEND WPE_PLATFORM_LIBRARIES nexus_client)
+else ()
+    message(STATUS "Building with nexus in proxy mode" )
+    list(APPEND WPE_PLATFORM_LIBRARIES nexus)
+endif ()
+
 list(APPEND WPE_PLATFORM_SOURCES
     src/bcm-nexus/renderer-backend.cpp
     src/bcm-nexus/view-backend.cpp
