diff -ruN WPEBackend-rdk.orig/cmake/FindWPE.cmake WPEBackend-rdk/cmake/FindWPE.cmake
--- WPEBackend-rdk.orig/cmake/FindWPE.cmake	2020-07-03 13:33:06.000000000 +0530
+++ WPEBackend-rdk/cmake/FindWPE.cmake	2021-01-12 12:47:50.586624784 +0530
@@ -29,7 +29,7 @@
 # ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 find_package(PkgConfig)
-pkg_check_modules(PC_WPE QUIET wpe-1.0)
+pkg_search_module(PC_WPE REQUIRED wpe-0.2 wpe-1.0)
 
 find_path(WPE_INCLUDE_DIRS
     NAMES wpe/wpe.h
@@ -37,7 +37,7 @@
 )
 
 find_library(WPE_LIBRARIES
-    NAMES wpe-1.0
+    NAMES wpe-0.2 wpe-1.0
     HINTS ${PC_WPE_LIBDIR} ${PC_WPE_LIBRARY_DIRS}
 )
 
