diff --git a/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp b/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp
index a6aeb9a..d83bd46 100644
--- a/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp
@@ -75,8 +75,8 @@
 #define MEDIA_MAX_FRAMERATE 30.0f
 #else
 // Assume hardware video decoding acceleration up to 8K@60fps for the rest of the cases.
-#define MEDIA_MAX_WIDTH 7680.0f
-#define MEDIA_MAX_HEIGHT 4320.0f
+#define MEDIA_MAX_WIDTH 3840.0f
+#define MEDIA_MAX_HEIGHT 2160.0f
 #define MEDIA_MAX_FRAMERATE 60.0f
 #endif
 
