diff --git a/SampleApp/include/SampleApp/SampleApplication.h b/SampleApp/include/SampleApp/SampleApplication.h
index 902907c6e2..f6520eca0c 100644
--- a/SampleApp/include/SampleApp/SampleApplication.h
+++ b/SampleApp/include/SampleApp/SampleApplication.h
@@ -158,7 +158,7 @@ public:
             avsCommon::sdkInterfaces::ChannelVolumeInterface::Type speakerType);
     };
 
-private:
+protected:
     /**
      * Initialize a SampleApplication.
      *
