diff --git a/build/BuildDefaults.cmake b/build/BuildDefaults.cmake
index 2e696e6a30..d98b355e04 100644
--- a/build/BuildDefaults.cmake
+++ b/build/BuildDefaults.cmake
@@ -4,7 +4,7 @@
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
 
 # Disallow out-of-source-builds.
-include(DisallowOutOfSourceBuilds)
+#include(DisallowOutOfSourceBuilds)
 
 # Setup default build options, like compiler flags and build type.
 include(BuildOptions)
