From 87c78eea2c4ac69c09947c8e49f8e8668640071c Mon Sep 17 00:00:00 2001
From: Francois Perrad <francois.perrad@gadz.org>
Date: Sun, 7 Dec 2025 10:15:27 +0100
Subject: [PATCH] missing include

Signed-off-by: Francois Perrad <francois.perrad@gadz.org>
Upstream: upstream is dead
---
 fileaccess.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fileaccess.c b/fileaccess.c
index 738889d..c03d5d6 100644
--- a/fileaccess.c
+++ b/fileaccess.c
@@ -4,6 +4,7 @@
 #include <string.h>
 #include <ctype.h>
 #include <sys/stat.h>
+#include <unistd.h>
 #include "ficl.h"
 
 #if FICL_WANT_FILE
-- 
2.43.0

