diff -urN bcm-refsw-20150619/nexus/build/os/linuxkernel/os_flags.inc bcm-refsw-20150619.new/nexus/build/os/linuxkernel/os_flags.inc
--- bcm-refsw-20150619/nexus/build/os/linuxkernel/os_flags.inc	2015-06-19 23:16:34.000000000 +0200
+++ bcm-refsw-20150619.new/nexus/build/os/linuxkernel/os_flags.inc	2015-11-03 22:00:47.644807608 +0100
@@ -113,7 +113,7 @@
 # Seperate MIPS and ARM here
 ifneq ($(filter ${B_REFSW_ARCH}, arm-linux), ${B_REFSW_ARCH})
 NEXUS_CFLAGS += -fomit-frame-pointer -G 0 -mno-abicalls -fno-pic -Wa,--trap -DMODULE -mlong-calls
-NEXUS_CFLAGS += -msoft-float -pipe
+NEXUS_CFLAGS += -mhard-float -mfloat-abi=hard -mfpu=neon-vfpv4 -march=armv7-a -mtune=cortex-a7 -pipe
 NEXUS_CFLAGS += -DLINUX -D__KERNEL__
 ifeq ($(LINUX_VER_GE_2_6_31),y)
 LINUX_ASM_INC ?= $(LINUX)/arch/mips/include/asm/mach-brcmstb
@@ -169,7 +169,7 @@
 
 NEXUS_CFLAGS+=  -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -Werror-implicit-function-declaration -Wno-format-security -fno-delete-null-pointer-checks -fno-dwarf2-cfi-asm -mabi=aapcs-linux -mno-thumb-interwork -funwind-tables -marm -D__LINUX_ARM_ARCH__=7 -march=armv7-a -Uarm -fno-stack-protector -fomit-frame-pointer -femit-struct-debug-baseonly -fno-var-tracking -Wdeclaration-after-statement -fno-strict-overflow -fconserve-stack -Werror=implicit-int -Werror=strict-prototypes -Wno-unused-but-set-variable
 # We are using -mfpu=vfpv3-d16 to force off generation of NEON instructions for fixed point code, when compiler comes with default -mfpu=neon-vfpv4
-NEXUS_CFLAGS += -msoft-float -mfloat-abi=softfp -mfpu=vfpv3-d16
+NEXUS_CFLAGS += -mhard-float -mfloat-abi=hard -mfpu=neon-vfpv4 -march=armv7-a -mtune=cortex-a7
 NEXUS_CFLAGS += -fno-pic
 endif
 
