diff -auNrd a/nexus/modules/sage/src/linuxuser/nexus_sage_bp3_bin_path.c a/nexus/modules/sage/src/linuxuser/nexus_sage_bp3_bin_path.c
--- a/nexus/modules/sage/src/linuxuser/nexus_sage_bp3_bin_path.c	2019-08-01 15:47:08.147945883 +0200
+++ a/nexus/modules/sage/src/linuxuser/nexus_sage_bp3_bin_path.c	2019-08-01 15:47:49.847933874 +0200
@@ -57,4 +57,4 @@
 **   This configuration variable holds the path where the BP3 bin file is on the filesystem.
 **   Customers can change the value of this variable to meet their needs.
 ******************************************************************************/
-char bp3_bin_file_path[] =".";
+char bp3_bin_file_path[] ="/usr/bin";
