--- BSEAV/lib/gpu/vc5/platform/nexus/display_nexus.c.orig	2019-01-23 17:21:13.301691128 -0500
+++ BSEAV/lib/gpu/vc5/platform/nexus/display_nexus.c	2019-01-23 17:22:37.941918452 -0500
@@ -271,7 +271,6 @@
 
 static void SetWindowInfo(WindowInfo *dst, NXPL_NativeWindowInfoEXT *src)
 {
-   static_assert(sizeof(WindowInfo) == sizeof(NXPL_NativeWindowInfoEXT), "sizeof(WindowInfo) & sizeof(NXPL_NativeWindowInfoEXT) need to match");
    dst->width = src->width;
    dst->height = src->height;
    dst->x = src->x;
