--- ./BSEAV/lib/gpu/vc5/platform/nexus/display_nexus.c.orig	2020-02-22 08:04:56.645613174 +0530
+++ ./BSEAV/lib/gpu/vc5/platform/nexus/display_nexus.c	2020-02-22 08:05:03.241560692 +0530
@@ -268,7 +268,6 @@
 
 static void SetWindowInfo(WindowInfo *dst, NXPL_NativeWindowInfoEXT *src)
 {
-   static_assert(sizeof(WindowInfo) == sizeof(NXPL_NativeWindowInfoEXT), "sizeof(WindowInfo) & sizeof(NXPL_NativeWindowInfoEXT) need to match");
    dst->width = src->width;
    dst->height = src->height;
    dst->x = src->x;
