From 77f1ad95beb4cef6a45861e97916d6a84dc8690b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Enrique=20Oca=C3=B1a=20Gonz=C3=A1lez?= <eocanha@igalia.com>
Date: Mon, 15 May 2017 10:38:57 +0000
Subject: [PATCH] qtdemux: select the preferred protection system even if not
 available

---
 gst/isomp4/qtdemux.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index 7b71237..bfbc2b5 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -7682,12 +7682,22 @@ gst_qtdemux_configure_protected_caps (GstQTDemux * qtdemux,
   }
 
   if (!selected_system) {
+    GST_WARNING_OBJECT(qtdemux, "selecting %s as preferred protection system even though "
+            "it wasn't among the available protection systems", qtdemux->preferred_protection_system_id);
+    const gchar *preferred_system_array[] =
+                { qtdemux->preferred_protection_system_id, NULL };
+    selected_system = gst_protection_select_system (preferred_system_array);
+  }
+
+  /*
+  if (!selected_system) {
     g_ptr_array_add (qtdemux->protection_system_ids, NULL);
     selected_system = gst_protection_select_system ((const gchar **)
         qtdemux->protection_system_ids->pdata);
     g_ptr_array_remove_index (qtdemux->protection_system_ids,
         qtdemux->protection_system_ids->len - 1);
   }
+  */
 
   if (!selected_system) {
     GST_ERROR_OBJECT (qtdemux, "stream is protected, but no "
-- 
1.8.3.2

