diff -ruN wpeframework-netflix.orig/CMakeLists.txt wpeframework-netflix/CMakeLists.txt
--- wpeframework-netflix.orig/CMakeLists.txt	2020-08-05 19:56:15.083212752 +0530
+++ wpeframework-netflix/CMakeLists.txt	2020-08-05 19:58:09.930282231 +0530
@@ -49,7 +49,6 @@
     endif()
 endif()
 if(NETFLIX_VERSION_5_2)
-    find_package(Harfbuzz REQUIRED)
     find_package(Freetype2 REQUIRED)
 
     target_compile_definitions(${MODULE_NAME}
@@ -58,8 +57,7 @@
 
     target_include_directories(${MODULE_NAME}
         PRIVATE
-            ${FREETYPE2_INCLUDE_DIRS}
-            ${HARFBUZZ_INCLUDE_DIRS})
+            ${FREETYPE2_INCLUDE_DIRS})
 
     target_compile_definitions(${MODULE_NAME}
     PRIVATE
