diff --git a/Source/core/IIterator.h b/Source/core/IIterator.h
index f75c059c..3d6236d6 100644
--- a/Source/core/IIterator.h
+++ b/Source/core/IIterator.h
@@ -162,7 +162,7 @@ namespace Core {
         {
             ASSERT(IsValid());
 
-            return (*m_Iterator);
+                return (*m_Iterator);
         }
 
         inline const ELEMENT& Current() const
@@ -209,6 +209,11 @@ namespace Core {
         {
             return (m_Container);
         }
+        inline void Container(CONTAINER& container) {
+            m_Container = &container;
+            m_Iterator = m_Container->begin();
+            m_Index = 0;
+        }
 
     private:
         CONTAINER* m_Container;
