diff -ruN wpeframework.orig/cmake/project.cmake.in wpeframework/cmake/project.cmake.in
--- wpeframework.orig/cmake/project.cmake.in	2021-03-17 02:41:05.000000000 +0530
+++ wpeframework/cmake/project.cmake.in	2021-03-18 12:30:33.221661460 +0530
@@ -236,6 +236,11 @@
                 "${CMAKE_INSTALL_PREFIX}/../etc/${NAMESPACE}/plugins/"
                 COMPONENT ${_component})
     endforeach()
+
+    unset(_plugins)
+    unset(_classname)
+    unset(_component)
+
 endmacro()
 
 function(add_compiler_flags modulename options)
