--- host-python-pycrypto-2.6.1/lib/Crypto/Random/_UserFriendlyRNG.py.orig	2020-05-16 07:08:47.408744999 -0400
+++ host-python-pycrypto-2.6.1/lib/Crypto/Random/_UserFriendlyRNG.py	2020-05-16 07:12:24.156048566 -0400
@@ -74,7 +74,10 @@
         self._time_es.feed(struct.pack("@I", int(2**30 * (t - floor(t)))))
 
         # Add the fractional part of time.clock()
-        t = time.clock()
+        try:
+            t = time.clock()
+        except AttributeError:
+            t = time.process_time()
         self._clock_es.feed(struct.pack("@I", int(2**30 * (t - floor(t)))))
 
 
