diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b663e4..529ac74 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -64,7 +64,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall")
 #-------------------------------------------------------------------------------
 find_package (Threads)
 
-if (NOT BUILD_YOCTO)
+if (NOT BUILD_YOCTO AND NOT BUILD_BR)
 
 # base64 external dependency
 #-------------------------------------------------------------------------------
